src/main/java/com/codegym/jrugotom5/service/AdvertService.java

```diff
@@ -38,5 +41,20 @@ public List<AdvertBasicInfoDTO> getAllAdverts() {
         return Streamable.of(advertRepository.findAll())
                 .map(advert->modelMapper.map(advert, AdvertBasicInfoDTO.class))
                 .toList();
+      
+    public List<AdvertInfoForCreatorDto> getAdvertsByCreateBy(Long id) {
+        try{
+            List<Advert> adverts = advertRepository.getAdvertsByCreatedBy_Id(id);
+            if(adverts.isEmpty()){
+                log.info("No adverts found for id {}", id);
+                return Collections.emptyList();
+            }
+            return adverts.stream()
+                    .map(advert -> modelMapper.map(advert, AdvertInfoForCreatorDto.class))
+                    .toList();
+        }catch (Exception e){
+            log.error(e.getMessage());
```

Comment:
if there really might be an exception, use an ExceptionHandler for that purpose

Reply:
> if there really might be an exception, use an ExceptionHandler for that purpose

and custom exception